servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -313,12 +311,16 @@ public boolean isDirectory(long id) {
    * Returns the file id for a given path. Called via RPC, as well as internal masters.
    *
    * @param path the path to get the file id for
-   * @return the file id for a given path
-   * @throws InvalidPathException
+   * @return the file id for a given path, or -1 if there is no file at that path
    */
-  public long getFileId(TachyonURI path) throws InvalidPathException {
+  public long getFileId(TachyonURI path) {
     synchronized (mInodeTree) {
-      Inode inode = mInodeTree.getInodeByPath(path);
+      Inode inode;
+      try {
+        inode = mInodeTree.getInodeByPath(path);
+      } catch (InvalidPathException e) {
+        return -1;
```

Comment:
Right now in theory, -1 is still a valid fileId that a block master may assign. Though I think the risk of running into this issue is low, I think it is necessary to have a TODO here to make sure client and master agrees on some impossible file ids.